networks/suzuka/suzuka-full-node/src/partial.rs

```diff
@@ -306,7 +307,17 @@ where
 			}
 			BlockCommitmentEvent::Rejected { height, reason } => {
 				debug!("Commitment rejected: {:?} {:?}", height, reason);
-				// TODO: block reversion
+				let current_head_height = executor.get_block_head_height().await?;
+				if height > current_head_height {
+					// Nothing to revert
+				} else {
+					match executor.revert_block_head(height - 1) {
+						Ok(_) => {}
+						Err(e) => {
+							error!("Failed to revert to block height: {:?}", e);
```

Comment:
This might actually be a case where we want to crash. If the node needs to revert but cannot, it will be costly for the operator and the network for it to continue operation. 

It's outside of the current scope, but... 

In cases like these, I think the most graceful thing to do would be to have the node optionally try and delegate before it goes offline. That way there's not a risk to the network of losing active stake and the node operator can continue to earn some rewards.

Reply:
Right, the quick solution is to propagate the error so that the node terminates.

> In cases like these, I think the most graceful thing to do would be to have the node optionally try and delegate before it goes offline.

I will need some pointers on how to do that. But I would treat an error here as an unrecoverable implementation bug.

Reply:
Made failures fatal in [350e002](https://github.com/movementlabsxyz/movement/pull/271/commits/350e00245c56917afd5cc0623e7b36dadb2e73b2)